chain/network/src/raw.rs

```diff
@@ -0,0 +1,345 @@
+use crate::network_protocol::{
+    Encoding, Handshake, HandshakeFailureReason, PartialEdgeInfo, PeerChainInfoV2, PeerIdOrHash,
+    PeerMessage, Ping, RawRoutedMessage, RoutedMessageBody,
+};
+use crate::time::Utc;
+use bytes::buf::{Buf, BufMut};
+use bytes::BytesMut;
+use near_crypto::{KeyType, SecretKey};
+use near_primitives::block::GenesisId;
+use near_primitives::hash::CryptoHash;
+use near_primitives::network::PeerId;
+use near_primitives::types::{AccountId, BlockHeight, EpochId};
+use near_primitives::version::{ProtocolVersion, PROTOCOL_VERSION};
+use std::io;
+use std::net::SocketAddr;
+use std::time::{Duration, Instant};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpStream;
+
+/// Represents a connection to a peer, and provides only minimal functionality.
+/// Almost none of the usual NEAR network logic is implemented, and the caller
+/// will receive messages via the recv() function. Currently the only message
+/// you can send is a Ping message, but in the future we can add more stuff there
+/// (e.g. sending blocks/chunk parts/etc)
+pub struct Peer {
+    my_peer_id: PeerId,
+    peer_id: PeerId,
+    secret_key: SecretKey,
+    stream: TcpStream,
+    buf: BytesMut,
+}
+
+/// The types of messages it's possible to receive from a `Peer`. Any PeerMessage
+/// we receive that doesn't fit one of these will just be logged and dropped.
+pub enum Message {
```

Comment:
if this enum is specifically representing received messaged, then rename it to sth like "Response".

Reply:
ok named it ReceivedMessage, since it's not always a Response (AccounceAccounts weren't requested by us)